rewrite-yaml/src/main/java/org/openrewrite/yaml/tree/JsonPathYamlVisitor.java

```diff
@@ -0,0 +1,232 @@
+/*
+ *  Copyright 2021 the original author or authors.
+ *  <p>
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  <p>
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *  <p>
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.openrewrite.yaml.tree;
+
+import org.openrewrite.Tree;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.yaml.internal.grammar.JsonPath;
+import org.openrewrite.yaml.internal.grammar.JsonPathBaseVisitor;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.BiPredicate;
+import java.util.stream.Collectors;
+
+@SuppressWarnings({"unchecked", "ConstantConditions"})
+public class JsonPathYamlVisitor extends JsonPathBaseVisitor<Object> {
```

Comment:
Since this visitor has internal state and is only ever used by the JsonPathMatcher, it might be better co-located with JsonPathMatcher (either as an inner class or a package private)

Reply:
👍 ✅